drivers/usb/device/usb_dc_stm32.c

```diff
@@ -263,6 +264,45 @@ static int usb_dc_stm32_clock_enable(void)
 	return 0;
 }
 
+#if defined(USB_OTG_FS) || defined(USB_OTG_HS)
+static u32_t usb_dc_stm32_get_maximum_speed(void)
+{
+	/*
+	 * If max-speed is not passed via DT, set it to USB controller's
+	 * maximum hardware capability.
+	 */
+#if defined(USB_HS_PHYC) && defined(CONFIG_USB_HS_BASE_ADDRESS)
+	u32_t speed = USB_OTG_SPEED_HIGH;
+#else
+	u32_t speed = USB_OTG_SPEED_FULL;
+#endif /* USB_HS_PHYC && CONFIG_USB_HS_BASE_ADDRESS */
+
+#ifdef CONFIG_USB_MAXIMUM_SPEED
```

Comment:
In Zephyr current state of the art, this code is fine.
In the evolution foreseen for device tree information usage, a clearer Kconfig/dts split is expected, along with fixup files removal.
Here this is clearly a case where we're using CONFIG_ #define which does not belong to Kconfig space but is purely device HW description, hence dts 'space'.

So, as a heads up, it is likely this piece will have to be reworked in some time, to make a clearer split between HW boot time configuration and device instantiation (device tree) and SW configuration (Kconfig).

Reply:
Thanks for the heads up. I will do the rework when the time comes.